DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/component_interfaces_comparison/csr-pipes/src/vector_add.cpp

```diff
@@ -0,0 +1,121 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class SimpleVAddPipes;
+
+// Forward declare pipe names to reduce name mangling (maybe get rid of; pipes
+// already have bad names)
+class IDPipeA;
+class IDPipeB;
+class IDPipeC;
+
+#define VECT_SIZE 256
+
+using PipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::ready_latency<0>));
+
+using InputPipeA =
+    sycl::ext::intel::experimental::pipe<IDPipeA, int, 0,
+                                         PipeProps>;
+using InputPipeB =
+    sycl::ext::intel::experimental::pipe<IDPipeB, int, 0,
+                                         PipeProps>;
+
+using CSRPipeProps = decltype(sycl::ext::oneapi::experimental::properties(
+    sycl::ext::intel::experimental::protocol_avalon_mm));
+
+// this csr pipe will only be read and write once with the total sum
```

Comment:
Why is it important to mention how many times it will be read from/written to?
```suggestion
// this csr pipe will only be read from and written to once
```